packages/nitro-protocol/.env

```diff
@@ -2,3 +2,4 @@ DEFAULT_GAS=6721975
 DEFAULT_GAS_PRICE=20000000000
 GANACHE_HOST=0.0.0.0
 GANACHE_PORT=8545
+NODE_ENV=dev
```

Comment:
Should this be in `.env.development?`?

Reply:
Yeah there shouldn't be a `.env` file checked in at all. Should only need `.env.example` and `.env.local` (or `.env.development`)

Reply:
At magmo, we were checking in `.env` and `.env.$ENVIRONMENT` files, but not `.env.local`or `.env.$ENVIRONMENT.local` files. I know this goes against the 12 factor app philosophy, but it means
- everyone had good defaults, including circleci. they can overwrite the defaults with their `.local` env files
- nobody had to remember to create their own `.env.` file, or figure out what to put in it

What's wrong with this approach?

Reply:
I guess one idea is that
- `devtools` exports a `config` object
- `devtools` is responsible for creating sensible defaults, depending on the node environment
- Any environment is able to overwrite these defaults with an environment variable
- No `.env` files are checked in

Maybe this is more like the 12FA approach?

Reply:
I'm not a fan of everyone needing to create their own `.env` either so +1 for that.

I don't think there's anything wrong with that. Given that the rules around picking up `.env.*` are pretty arbitrary, it's essentially our choice of whether we want

- `.env` to be the expected configuration for _only_ prod and have `.env.<X>` for other contexts or
- `.env` being the local configuration, custom configuration can be provided via `.env.<X>`, and prod context can be `.env.prod` (or something like that)

If anything I lean towards the latter as well.

Reply:
> Any environment is able to overwrite these defaults with an environment variable

It seems to me that `.env.<environment>` would be doing something similar to this. It'd be providing its context-specific _default_ env vars, and if even _these_ defaults need to be overwritten then they can be specified via env vars.

If that's not how you're thinking about it then I guess you could elaborate on what you're envisioning.

Reply:
what I mean by an "environment variable" is something that ends up on `process.env`.

Suppose I want to know the configured value of `VAR`. Currently, the `configEnvVariables` sets `process.env.VAR` to be the value of the shell variable `$VAR` if it exists. Otherwise, it looks for a `VAR` in this set of files: https://github.com/bkeepers/dotenv#what-other-env-files-can-i-use

It's modified to suit our monorepo by looking in the monorepo root as well: https://github.com/statechannels/monorepo/blob/master/packages/devtools/src/config/env.ts#L23-L27

I think this logic currently doesn't make sense, since I believe that `packages/nitro-protocol/.env` takes precedence over `.env.production`.